src/Illuminate/Database/Schema/Grammars/MySqlGrammar.php

```diff
@@ -492,15 +492,23 @@ public function compileSpatialIndex(Blueprint $blueprint, Fluent $command)
      */
     protected function compileKey(Blueprint $blueprint, Fluent $command, $type)
     {
-        return sprintf('alter table %s add %s %s%s(%s)',
+        $columns = collect($command->columns)->map(function (string $column) {
+            return self::isFunctionalExpression($column) ? $column : $this->wrap($column);
+        })->implode(', ');
+
+        return sprintf(
+            'alter table %s add %s %s%s(%s)',
             $this->wrapTable($blueprint),
             $type,
             $this->wrap($command->index),
-            $command->algorithm ? ' using '.$command->algorithm : '',
-            $this->columnize($command->columns)
+            $command->algorithm ? ' using ' . $command->algorithm : '',
+            $columns
         );
     }
-
+protected static function isFunctionalExpression(string $column): bool
+{
+    return preg_match('/\(.+\)/', $column);
+}
```

Comment:
Oh, sorry, forgot the indent 🤦🏻‍♂️😅 
```suggestion

    protected static function isFunctionalExpression(string $column): bool
    {
        return preg_match('/\(.+\)/', $column);
    }
```